### PR TITLE
Various new features

### PR DESCRIPTION
**Pull request by pikiou**
New features, which you can have a look at [here](https://pikiou.github.io/sc2-planner) :
- keyboard shortcuts to undo/redo anything that usually changes the url
- resource graphs for minerals, gas, supply, and race specific boosts
- more options for maxing worker optimization
- maxing MULE, Nexus chronos or injects optimization
- build order read by synthed voice with rudimental visuals. The reading can be controlled by voice

### CODE REVIEW

**Comment by jtsauls** [comment]
Wow these are really useful changes. The graphs and undo/redo alone make putting together a build much easier. Voice is amazing. Haven't tested the optimization features yet. Awesome job!

File: src/components/WebPage.tsx [hunk 1]
```diff
@@ -142,17 +148,60 @@ export default withRouter(
...
+
+        redo(): void {
+            if (this.historyPosition === this.history.length) {
+                return
+            }
+            const save = this.history[this.historyPosition]
+            this.historyPosition++
+            this.restoreSave(save)
+        }
+
```
Comment (BurnySc2) [inline src/components/WebPage.tsx:190]: So this only saves a browser history entry if a new race is selected (= build order is cleared)?

**Review by BurnySc2 — APPROVED** [review]
You are talking about keyboard shortcuts. Could explain which they are and in short what they do?

Otherwise great work!
With the statistics the user now has a good oversight about how much money he is floating or how much supply is left, and it's even visualized over time so you don't have to inpect each element.

**Comment by BurnySc2** [comment]
After playing around with it, if you have the multi-line build order open, the drag and drop functionality doesn't work properly.
Works well with single-row mode though.

**Comment by pikiou** [comment]
> You are talking about keyboard shortcuts. Could explain which they are and in short what they do?

The keyboard shortcuts are meant to undo/redo any change to the BO (and possibly the settings).


> After playing around with it, if you have the multi-line build order open, the drag and drop functionality (to re-order build order items) doesn't work properly.

Indeed, and for multi-line drag&drop we need to use another d&d library but I didn't feel like changing it ><


> And the tooltips from the statistics don't seem to disappear as soon as you leave the graph.

I just fixed it.

**Pull request merged**